### PR TITLE
Blackfin nce

### PR DESCRIPTION
**Pull request by thattil**
- nCe made high before programming blackfin to ensure seamless programming
- nCe neednt be touched for just resetting fpga

### CODE REVIEW

File: slsDetectorServers/slsDetectorServer/src/programViaBlackfin.c [hunk 1]
```diff
@@ -29,8 +32,11 @@
 #define CMD_GPIO9_DEFINE_IN "echo in > /sys/class/gpio/gpio9/direction"
 #define CMD_GPIO3_DEFINE_IN "echo in > /sys/class/gpio/gpio3/direction"
 
+// nConfig
 #define CMD_GPIO9_DONT_TOUCH_FLASH "echo 0 > /sys/class/gpio/gpio9/value"
-#define CMD_GPIO3_DONT_TOUCH_FLASH "echo 0 > /sys/class/gpio/gpio3/value"
+// nCE
+#define CMD_GPIO3_DONT_TOUCH_FLASH "echo 1 > /sys/class/gpio/gpio3/value"
```
Comment (alejandrohomsp) [inline slsDetectorServers/slsDetectorServer/src/programViaBlackfin.c:38]: Hi @thattil 

I am interested in understanding the origin of this change. In particular, I would like to know if it is exclusive of Jungfrau FW v2_3, released by @lopez-c on April, or if it applies as well to v2_2.

**Comment by lopez-c** [comment]
Hello @alejandrohomsp 

This change is not related to any firmware version, but to the kernel version running on the Blackfin of your Jungfrau readout board, in particular for the kernel released generated on 29/10/2021.

In particular, this change fixes a bug that could lead to a failure during the update of the readout board firmware.

**Comment by alejandrohomsp** [comment]
Hi @lopez-c 

Thanks a lot for the answer!

**Comment by thattil** [comment]
Hi @alejandrohomsp 

Sorry for the delay in replying.

Just in case you have not been informed, you can update your kernel using the updatekernel command. You can also get your kernel version using kernelversion command. They should cause less-to-none programming errors in the future (especially the error that said the fpga is taking more than 10 s to say it is programmed). I would recommend updating the kernel version of your Jungfrau blackfins if their kernel versions are older than 29.10.2021.

**Comment by alejandrohomsp** [comment]
Hi @thattil 

Thanks a lot for the information! Yes, as far as I remember, we updated our Jungfrau modules with the latest kernel :)

**Comment by thattil** [comment]
Hi @alejandrohomsp 

That's great! But just so you know, its useful only after this commit, ie. from the developer branch (and not 6.1.1).

Cheers,
Dhanya

**Pull request merged**